src/store/modules/target.js

```diff
@@ -1385,6 +1385,50 @@ export default ({
             return false
           })
       }
+    },
+    addTargetToServer ({ state, rootState, getters }, target) {
+      if (rootState.auth.authentication_token !== '') {
+        return axios.post('/targets', getters.mapTargetFromLocalToServer(target))
+          .then(function (response) {
+            target.id = response.data.id
+            state.targetListStore.push(target)
+            return true
+          })
+          .catch(function (response) {
+            return false
+          })
+      }
+    },
+    updateTargetToServer ({ state, rootState, getters, commit }, target) {
+      if (rootState.auth.authentication_token !== '') {
+        return axios.put('/targets/' + target.id, getters.mapTargetFromLocalToServer(target))
+          .then(function (response) {
+            commit('updateTarget', target)
+            return true
+          })
+          .catch(function (response) {
+            return false
+          })
+      }
+    },
+    clearTargetEntitiesToDeleteToServer ({ state, commit, rootState }) {
+      let promiseResult
+      state.entitiesToDelete.forEach(entity => {
```

Comment:
instead delete one by one, is better if we add an endpoint in the backend that is going to receive the list of target to delete to improve the performance, the same for delete multiples agents